src/shared/avl-consumer/utils.ts

```diff
@@ -1,16 +1,42 @@
+import { z } from "zod";
 import { SubscriptionIdNotFoundError } from "../avl/utils";
-import { queryDynamo } from "../dynamo";
+import { getDynamoItem, queryDynamo, scanDynamo } from "../dynamo";
 import {
     AvlConsumerSubscription,
     AvlSubscriptionStatus,
     avlConsumerSubscriptionSchema,
     avlConsumerSubscriptionsSchema,
 } from "../schema";
+import { createStringLengthValidation } from "../validation";
 
-export const getAvlConsumerSubscriptions = async (tableName: string, status: AvlSubscriptionStatus) => {
-    const subscriptions = await queryDynamo<AvlConsumerSubscription>({
+export const subscriptionTriggerMessageSchema = z.object({
+    subscriptionPK: z.string(),
+    SK: z.string(),
+    frequencyInSeconds: z.union([z.literal(10), z.literal(15), z.literal(20), z.literal(30)]),
+    queueUrl: z.string(),
+});
+
+export type AvlSubscriptionTriggerMessage = z.infer<typeof subscriptionTriggerMessageSchema>;
+
+export const subscriptionDataSenderMessageSchema = z
+    .string()
+    .transform((body) => JSON.parse(body))
+    .pipe(
+        z.object({
+            subscriptionPK: createStringLengthValidation("subscriptionPK"),
+            SK: createStringLengthValidation("SK"),
+        }),
+    );
+
+export type AvlSubscriptionDataSenderMessage = z.infer<typeof subscriptionDataSenderMessageSchema>;
+
+export const getAvlConsumerSubscriptionsByStatus = async (tableName: string, status: AvlSubscriptionStatus) => {
+    const subscriptions = await scanDynamo<AvlConsumerSubscription>({
```

Comment:
Is there a specific reason for using Scan over RecursiveScan? 

My worry is that if we have loads of consumer subscriptions we might get to a point where we reach the size limit of what a scan could return which is why we use RecursiveScan in other places in the code base.

Reply:
No particular reason. Can you explain the recursive scan in more detail? I wasn't aware that's what its purpose was